byzcoin/statetrie.go

```diff
@@ -0,0 +1,177 @@
+package byzcoin
+
+import (
+	"encoding/binary"
+	"errors"
+
+	bolt "github.com/coreos/bbolt"
+	"github.com/dedis/cothority/byzcoin/trie"
+	"github.com/dedis/cothority/darc"
+)
+
+var errKeyNotSet = errors.New("key not set")
+
+// ReadOnlyStateTrie is the read-only interface for StagingStateTrie and
+// StateTrie.
+type ReadOnlyStateTrie interface {
+	GetValues(key []byte) (value []byte, contractID string, darcID darc.ID, err error)
+	GetProof(key []byte) (*trie.Proof, error)
+}
+
+// StagingStateTrie is a wrapper around trie.StagingTrie that allows for use in
+// byzcoin.
+type StagingStateTrie struct {
```

Comment:
Please consider what's the minimum that can be exported. Maybe just ReadOnlyStateTrie and make StagingStateTrie and StateTrie private so as not to clutter the public API of pkg byzcoin.